lib/matplotlib/axes/_axes.py

```diff
@@ -4147,6 +4147,10 @@ def bxp(self, bxpstats, positions=None, widths=None, vert=True,
         --------
         .. plot:: gallery/statistics/bxp.py
         """
+        if medianprops is None:  # Clamp median line to edge of box
+            medianprops = {"solid_capstyle": "butt"}
+        else:
+            medianprops.setdefault("solid_capstyle", "butt")
```

Comment:
Shouldn't use `setdefault` as that will modify the user's input dictionary. Something like this should avoid that:
```suggestion
        # Clamp median line to edge of box by default.
        medianprops = {
            "solid_capstyle": "butt",
            **(medianprops or {}),
        }
```

Reply:
Ooops, `setdefault` was my suggestion.  Of course you’re right.

Reply:
Just for my understanding, why exactly shouldn't we use setdefault? If the key exists in the dictionary, then setdefault won't make any changes. Am I understanding this correctly?

Reply:
Good question.

Suppose a user has a dictionary of properties that they re-use for multiple different plot types in their script.  If that dictionary gets added to when they pass it to `boxplot`, then it could affect other plots that are created further down the script.  That would be unexpected (and therefore undesirable) behaviour.

Does that make sense?  You are right that `setdefault` won’t change the dictionary if it already has that key.